Code/Framework/AzToolsFramework/AzToolsFramework/Metadata/MetadataManager.cpp

```diff
@@ -24,6 +25,8 @@ namespace AzToolsFramework
             // Note the version here is not the same as the MetadataFile version, since this is a separate serialization
             serializeContext->Class<MetadataManager, AZ::Component>()->Version(1);
         }
+
+        UuidEntry::Reflect(context);
```

Comment:
Why are you calling UuidEntry::Reflect from within MetadataManager::Reflect, instead of in the function that calls MetadataManager::Reflect?

Reply:
I guess it doesn't make sense to have it here, but this reflect is called as part of the component setup stuff (GetRequiredComponents) so I can't reflect non-components there.  But this is required by the UuidUtilComponent so Ill move the reflect into that call instead